### PR TITLE
APP-908 Implement an engine to build the EntityJSON according to the metadata

### PR DESCRIPTION
**Pull request by robson-sanchez**
- Added support for multipart form serialization (besides JSON) on API clients.
- Working with Message object (in lieu of plain string) to support messageML v1 and v2.
- Helper class to retrieve Spring beans to be used for non-component classes.

### CODE REVIEW

File: integration-commons/src/main/java/org/symphonyoss/integration/utils/ApplicationContextUtils.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+   * @param type type the bean must match; can be an interface or superclass.
+   * @return an instance of the single bean matching the required type
+   * @throws NoSuchBeanDefinitionException if no bean of the given type was found or context is null
+   */
+  public static <T> T getRequiredBean(Class<T> type) {
+    if (context != null) {
+      return context.getBean(type);
+    }
+
+    throw new NoSuchBeanDefinitionException(type);
```
Comment (dnsymphony) [inline integration-commons/src/main/java/org/symphonyoss/integration/utils/ApplicationContextUtils.java]: This should probably be IllegalStateException.

File: integration-commons/src/main/java/org/symphonyoss/integration/utils/ApplicationContextUtils.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+   * isn't provided
+   */
+  public static <T> T getBean(Class<T> type) {
+    if (context != null) {
+      try {
+        return context.getBean(type);
+      } catch (BeansException e) {
+        return null;
+      }
+    }
```
Comment (dnsymphony) [inline integration-commons/src/main/java/org/symphonyoss/integration/utils/ApplicationContextUtils.java]: If context IS null, it should throw IllegalStateException

File: integration-commons/src/main/java/org/symphonyoss/integration/utils/ApplicationContextUtils.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+/**
+ * Helper class to retrieve Spring beans to be used for a non-component classes.
+ * Created by rsanchez on 10/02/17.
+ */
+@Component
+public class ApplicationContextUtils {
```
Comment (dnsymphony) [inline integration-commons/src/main/java/org/symphonyoss/integration/utils/ApplicationContextUtils.java]: This smells of bad design.  Why is it needed?  What non-Spring-managed classes need to access application context?

File: integration-commons/src/main/java/org/symphonyoss/integration/event/HealthCheckServiceEvent.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package org.symphonyoss.integration.event;
+
+/**
+ * Event object to perform the health check in an specific service.
+ *
+ * Created by rsanchez on 22/03/17.
+ */
+public class HealthCheckServiceEvent {
```
Comment (dnsymphony) [inline integration-commons/src/main/java/org/symphonyoss/integration/event/HealthCheckServiceEvent.java]: What makes this an Event?  Same with the following class MessageMLVersionUpdatedEvent. They don't implement an Event interface.  They just seem to be wrapping a single field.

**Review by dnsymphony — CHANGES_REQUESTED** [review]

File: integration-commons/src/main/java/org/symphonyoss/integration/event/HealthCheckServiceEvent.java (hunk 4 above)
Comment (robson-sanchez) [reply integration-commons/src/main/java/org/symphonyoss/integration/event/HealthCheckServiceEvent.java]: Renamed the classes to HealthCheckEventData and MessageMLVersionUpdatedEventData respectively.

File: integration-commons/src/main/java/org/symphonyoss/integration/utils/ApplicationContextUtils.java (hunk 3 above)
Comment (robson-sanchez) [reply integration-commons/src/main/java/org/symphonyoss/integration/utils/ApplicationContextUtils.java]: I agree with you @dnsymphony, it seems a bad design. I'm removing this class and refactoring the code was using this one.

**Comment by robson-sanchez** [comment]
@dnsymphony please check again

**Review by dnsymphony — APPROVED** [review]

**Pull request merged**